src/gui/editorwidgets/qgsdoublespinbox.h

```diff
@@ -156,6 +163,13 @@ class GUI_EXPORT QgsDoubleSpinBox : public QDoubleSpinBox
     bool mExpressionsEnabled = true;
 
     QString stripped( const QString &originalText ) const;
+
+    // This is required because private implementation of
+    // QAbstractSpinBoxPrivate checks for specialText emptiness
+    // and skips specialText handling if it's empty
+    static QString SPECIAL_TEXT_WHEN_EMPTY;
```

Comment:
Any special reason to put this into the header?

Reply:
Do you mean the comment?

Reply:
No, the whole variable appears like an internal implementation detail that could only live isolated in the cpp on first look?

Reply:
Well, besides that I'm using `SPECIAL_TEXT_WHEN_EMPTY` in the test, I thought it could be clearer to define a variable for that char instead of hardcoding the utf code directly (if is that what you meant). 
But I'm fine to change it if you insist :)

Reply:
I don't insist, I was just thinking out loud. Mostly I try to keep implementation details out of installed headers.
Do as you wish ;)